catalog/core/catalog-core-impl/filter-proxy/src/main/java/ddf/catalog/filter/proxy/adapter/GeotoolsFilterAdapterImpl.java

```diff
@@ -134,6 +135,22 @@ public Object visit(Function expression, Object delegate) {
             throw new UnsupportedOperationException(
                     Function.class.getSimpleName() + " must not be null.");
         }
+
+        if (expression.getName().equals(ProximityFunction.NAME.getName())) {
+            List<Expression> parameters = expression.getParameters();
+            if (parameters.size() == 3) {
```

Comment:
The number of parameters of the function should only be 3; the query should fail otherwise.  Updated this to be a constant in ProximityFunction